vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/embedding_processing_unittest.cc

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+
+#include <map>
+#include <utility>
+#include <vector>
+
+#include "base/files/file.h"
+#include "bat/ads/internal/base/unittest/unittest_base.h"
+#include "bat/ads/internal/base/unittest/unittest_file_util.h"
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+
+// npm run test -- brave_unit_tests --filter=BatAds*
+
+namespace {
+constexpr char kSimpleResourceFile[] =
+    "wtpwsrqtjxmfdwaymauprezkunxprysm_simple";
+}  // namespace
+
+namespace ads {
+
+using testing::_;
+using testing::Invoke;
+
+class BatAdsEmbeddingProcessingPipelineTest : public UnitTestBase {
+ protected:
+  BatAdsEmbeddingProcessingPipelineTest() = default;
+
+  ~BatAdsEmbeddingProcessingPipelineTest() override = default;
+};
+
+TEST_F(BatAdsEmbeddingProcessingPipelineTest, EmbedTextSimple) {
+  // Arrange
+  resource::TextEmbedding resource;
+  EXPECT_CALL(*ads_client_mock_, LoadFileResource(_, _, _))
+      .WillOnce(Invoke([](const std::string& /*id*/, const int /*version*/,
+                          LoadFileCallback callback) {
+        const base::FilePath path =
+            GetFileResourcePath().AppendASCII(kSimpleResourceFile);
+
+        base::File file(
+            path, base::File::Flags::FLAG_OPEN | base::File::Flags::FLAG_READ);
+        std::move(callback).Run(std::move(file));
+      }));
```

Comment:
`LoadFileResource` is already mocked in `vendor/bat-native-ads/src/bat/ads/internal/base/unittest/unittest_mock_util.cc`, please override `SetupMocks` use `CopyFileFromTestPathToTempPath` in `SetupMocks`. See `vendor/bat-native-ads/src/bat/ads/internal/base/unittest/README.md` for more info.